src/main/groovy/com/rundeck/plugins/ansible/ansible/AnsibleInventoryBuilder.java

```diff
@@ -25,6 +26,9 @@ public File buildInventory() throws ConfigurationException {
             PrintWriter writer = new PrintWriter(file);
             AnsibleInventory ai = new AnsibleInventory();
             for (INodeEntry e : nodes) {
+                if( e.getHostname() == null ){
+                    throw new MissingPropertyException("No hostname for node: " + e.getNodename());
```

Comment:
I think you should throw ConfigurationException.  The `catch(Exception e)` maybe should just do `catch(IOException e)`